packages/manager/apps/pci-public-ip/src/api/hooks/useImportIP.ts

```diff
@@ -0,0 +1,46 @@
+import { PaginationState } from '@ovhcloud/manager-components';
+import { useQuery, useMutation } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { paginateResults } from '@/api/utils/pagination';
+import { getImportsIPs, moveIPToProject } from '../data/import-ip';
+
+export type ImportsOptions = {
+  pagination: PaginationState;
+};
+
+export const useGetImportsIPs = (
+  projectId: string,
+  { pagination }: ImportsOptions,
+) => {
+  const { data, error, isLoading } = useQuery({
+    queryKey: ['ip', 'type', 'failover'],
+    queryFn: () => getImportsIPs(projectId),
+  });
+
+  return useMemo(() => {
+    return {
+      isLoading,
+      error,
+      data: paginateResults(data || [], pagination),
```

Comment:
you can also pass pagination directly to iceberg fetch so you don't have to paginate in the frontend

Reply:
Done